internal/rbd/controllerserver.go

```diff
@@ -519,6 +519,11 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 
 		// All errors other than ErrImageNotFound should return an error back to the caller
 		case ErrImageNotFound:
+			err = rbdVol.removeImageFromTrash()
+			if err != nil {
+				klog.Errorf(util.Log(ctx, "failed to delete rbd image: %s from trash with error: %v"), rbdVol, err)
+				return nil, status.Error(codes.Internal, err.Error())
```

Comment:
Is this a critical error? Is the trash not cleaned up at a later time automatically?
Deleting an image should probably succeed when the image is gone (which could be in the trash).
I think logging the error (and mentioning the trash is cleared later) is sufficient?

Reply:
Removed now, lets ceph handle it, if required this can be added later also